openrr-teleop/src/control_nodes_config.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{
+    ControlNode, IkNode, IkNodeConfig, JoyJointTeleopNode, JoyJointTeleopNodeConfig, MoveBaseNode,
+};
+use arci::{JointTrajectoryClient, MoveBase, Speaker};
+use openrr_client::IkSolverWithChain;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct JoyJointTeleopConfig {
+    pub config: JoyJointTeleopNodeConfig,
+    pub client_name: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct IkNodeTeleopConfig {
+    pub config: IkNodeConfig,
+    pub solver_name: String,
+    pub joint_trajectory_client_name: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ControlNodesConfig {
+    pub joy_joint_teleop_configs: Vec<JoyJointTeleopConfig>,
+    pub move_base_mode: Option<String>,
+    pub ik_node_teleop_configs: Vec<IkNodeTeleopConfig>,
+}
+
+impl ControlNodesConfig {
+    pub fn create_control_nodes(
```

Comment:
Comment:
* member functionn
* function with argument

Uhhm. which is better?